Feature/update

alfondotnet opened the pull request.

ljharb committed:
[eslint config] [base] [breaking] re-disable `import/extensions` rule

ref https://github.com/benmosher/eslint-plugin-import/issues/390

ljharb committed:
[eslint config] [base] add some `import` plugin settings

ljharb committed:
[eslint config] [base] [patch] `newline-per-chained-call`: bump the limit to 4.

Specifically, this is to allow this frequent pattern:
```js
$.ajax({
  …
}).done(() => {
  …
}).fail(() => {
  …
}).always(() => {
  …
});
```

ljharb committed:
[eslint config] [base] [Deps] update `eslint-plugin-import`, `tape`, `eslint-find-rules`

ljharb committed:
[eslint config] [breaking] [base] update `eslint` to v3

 - drop support for node 4.

ljharb committed:
[eslint config] [base] v4.0.0

cmalard committed:
Remove leading underscores for private properties

- It follows [ES6 styleguide](https://github.com/airbnb/javascript#naming--leading-underscore)
- eslint plugin is configured this way : `'no-underscore-dangle': [2, { allowAfterThis: false }],`

lencioni committed:
Add import/no-restricted-paths to imports rules

This rule was added by 1.10.0. I am setting it to be disabled because it
is very project-specific, so it doesn't make sense to enable it here.

https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-restricted-paths.md

I found the sections of this file a little confusing, so I didn't know
where to put it. I think it would be better to simply alphabetize the
rules but I'll leave that for another time.

lencioni committed:
Allow importing devDependencies in tests

This rule seems to be giving me errors for some places we have imported
devDependencies in our tests such as tape and eslint. This is actually
okay, so we just need to configure this rule to allow devDependencies
for tests.

While I was at it, I added some whitespace to improve readability and
reduce the likelihood of merge conflicts.

ljharb committed:
[eslint config] [*] [Tests] stop testing < node 4.

ljharb committed:
[eslint config] [deps] [breaking] update `eslint` to v3, `eslint-config-airbnb-base` to v5, `eslint-find-rules`, `eslint-plugin-import`, `eslint-plugin-jsx-a11y` to v2, `eslint-plugin-react` to v6, `tape`.

 - drop node < 4 support

ljharb committed:
[eslint config] [react] [breaking] enable react rules:

 - `react/jsx-filename-extension`
 - `react/jsx-no-comment-textnodes`
 - `react/jsx-no-target-blank`
 - `react/require-extension`
 - `react/no-render-return-value`
 - `react/no-find-dom-node`
 - `react/no-deprecated`

ljharb committed:
[eslint config] [react] [breaking] enable aria rules:

 - `jsx-a11y/aria-props`
 - `jsx-a11y/aria-proptypes`
 - `jsx-a11y/aria-unsupported-elements`
 - `jsx-a11y/role-has-required-aria-props`
 - `jsx-a11y/role-supports-aria-props`

ljharb committed:
[eslint config] [react] [breaking] enable jsx-a11y rules:

 - `jsx-a11y/heading-has-content`
 - `jsx-a11y/html-has-lang`
 - `jsx-a11y/lang`
 - `jsx-a11y/no-marquee`
 - `jsx-a11y/scope`
 - `jsx-a11y/href-no-hash`
 - `jsx-a11y/label-has-for`

cmalard committed:
Document for-in restricted usage

https://github.com/airbnb/javascript/issues/851#issuecomment-213834028

ljharb committed:
[guide] [es5] Remove ES5 guide and mark it deprecated.

Fixes #941.

LukeL99 committed:
Added link to Mozilla documentation on Function.bind

ljharb committed:
[guide] Add link to `eol-last` rule.

Fixes #945.

Commit:
Changed ESLint rule configs to use 'off', 'warn', and 'error' instead of numbers for better readability

ljharb committed:
[eslint config] [base] [fix] Prevent trailing commas in the legacy config.

Fixes #950.

ljharb committed:
[eslint config] [base] [breaking] enable `import/newline-after-import`

Per https://github.com/benmosher/eslint-plugin-import/issues/386 / https://github.com/benmosher/eslint-plugin-import/issues/441

ljharb committed:
[eslint config] [base] [deps] update `eslint-plugin-import`

ljharb committed:
[eslint config] [base] v4.0.1

ljharb committed:
[eslint config] [base] Fleshing out defaults.

ljharb committed:
[eslint config] [base] [breaking] enable overlooked rules:

 - `linebreak-style`
 - `new-parens`
 - `no-continue`
 - `no-lonely-if`
 - `operator-assignment`
 - `space-unary-ops`
 - `dot-location`
 - `no-extra-boolean-cast`
 - `no-this-before-super`
 - `require-yield`
 - `no-path-concat`
 - `no-label-var`

ljharb committed:
[eslint config] [base] [fix] repair accidental comma-dangle change.

ljharb committed:
[eslint config] [base] v4.0.2

Commit:
Capitalize all comments in react.js

ronal2do committed:
Airbnb React/JSX Style Guide translation for PT-br

ronal2do committed:
Airbnb React/JSX Style Guide translation for PT-BR

ljharb committed:
[eslint config] [base] [deps] update `eslint`, `eslint-find-rules`, `eslint-plugin-import`

ljharb committed:
Merge pull request #955 from ronal2do/master

Airbnb React/JSX Style Guide translation for PT-BR

ljharb committed:
[eslint config] [*] [docs] update the peer dep install command to dynamically look up the right version numbers when installing peer deps.

Fixes #956.

This command would be simpler if https://github.com/npm/npm/issues/11213 is ever implemented.

ljharb committed:
[eslint config] [*] [docs] forgot “—save-dev” in 623dc44bb9990dd90c0540be8309c7b5a3f6217a

ljharb committed:
[eslint config] [*] [tests] fix prepublish scripts.

ljharb committed:
[eslint config] [base] [breaking] enable more overlooked rules:

 - `no-void`
 - `constructor-super`
 - `prefer-spread`
 - `no-new-require`
 - `no-undef-init`

idugcoal committed:
change 'conjunctions' to 'contractions'

lencioni committed:
Merge pull request #961 from idugcoal/master

Change 'conjunctions' to 'contractions'

ljharb committed:
[eslint config] [*] [deps] update `babel-tape-runner`, add `babel-preset-airbnb`

ljharb committed:
[eslint config] [base] [breaking] enable `no-unexpected-multiline` rule

ljharb committed:
[eslint config] [base] v5.0.0

ljharb committed:
[eslint config] [base] `no-unused-expressions`: flesh out options

Fixes #963.

lencioni committed:
Add guideline for displayName in higher-order components

This guideline will help us keep our higher-order components consistent
and easy to understand.

Fixes #968.

jglover committed:
[eslint config] [*] [docs] Added subshell to usage commands and converted samples to the three backtick markdown syntax with language specifier for highlighting.

lencioni committed:
Add missing closing paren in HOC guideline

As pointed out by eagle-eyed @pepakriz.

lencioni committed:
Improve React HOC example

I decided that `WrappedComponent` is clearer than `Component` here, so I
made the switch. I also realized that `WrappedComponent.name` might
still be undefined, so I added a fallback value of "Component".

ljharb committed:
[eslint config] [base] [deps] update `eslint-plugin-import`

ljharb committed:
[eslint config] [base] [deps] update `eslint` to `v3.2`

ljharb committed:
[eslint config] [*] improve prepublish script

ljharb committed:
[eslint config] [base] v5.0.1

zanemountcastle committed:
Change 'Brazilian' to 'Portuguese'

ljharb committed:
[eslint config] v10.0.0

ljharb committed:
[eslint config] [*] add `safe-publish-latest` to `prepublish`

ljharb committed:
[guide] Use `var` in reserved word examples, since they refer to untranspiled code.

lencioni committed:
Add curlies around ref function body

The example we have here uses an implicit return, which is forbidden
when assigning. This triggers the no-return-assign rule.

Fixes #980

felipethome committed:
[guide] Add eslint rule to spacing in a function signature topic

[guide] Add eslint rules to spacing in a function signature topic

felipethome committed:
[guide] Add eslint rules to function declaration and function constructor topics

lencioni committed:
Remove rules about reserved words

In 53b4173b we removed the ES5 guide which contains a lot of guidelines
that are no longer very relevant for us. Similarly, some folks have
raised the relevance of these rules about reserved words, given that we
are now living in an age where ES3 support has mostly waned and
transpilers such as Babel are widely adopted and pave over these issues.
This seems like a good opportunity to simplify.

Fixes #61

lencioni committed:
Add template literal "bad" example to single quotes guideline

There has been some confusion around whether we should use single quotes
or template literals. To help avoid this confusion, I am adding a "bad"
example to the single quotes guideline. This rule is already enforced by
the quotes linter rule.

Generally, we want code to clearly express developer intention. Using
template literals communicates that you intend to use some of the
features that template literals offer (e.g. interpolation).

Fixes #992

lencioni committed:
Reverse rule on string concatenation for long lines (#995)

Broken and concatenated long strings are painful to work with and
produce less readable and searchable code. I think we should reverse
this rule.

Unfortunately, the max-len rule currently does not allow for this, but
there is currently a proposal to add an option to ESLint's max-len rule
that would allow for strings to be ignored.

  https://github.com/eslint/eslint/issues/5805

There have also been discussions around performance of string
concatenation (https://github.com/airbnb/javascript/issues/40), but I
don't think that is very relevant here so I removed the links to them.

felipethome committed:
[guide] Add ESlint rules to 'always use const' and 'decriptive variable names' topics

felipethome committed:
[guide] Add a topic about spread operator vs apply

felipethome committed:
[guide] Simplify the examples of 'apply vs spread' topic

eterey committed:
Update README.md

mitsuruog committed:
Add link for Japanese translation

ljharb committed:
[eslint config] [*] [deps] update `eslint`, `eslint-find-rules`

ljharb committed:
[eslint config] [base] [deps] update `eslint`, `eslint-plugin-import`

 - add temp disabled `func-call-spacing` rule
 - add temp disabled `no-template-curly-in-string` rule
 - enable `no-global-assign` and deprecate `no-native-reassign`
 - enable `no-unsafe-negation` and deprecate `no-negated-in-lhs`
 - add disabled `sort-keys` rule

ljharb committed:
[eslint config] [base] v5.0.2

ljharb committed:
[eslint config] [deps] update `eslint`, `eslint-plugin-jsx-a11y`, `eslint-plugin-import`, `eslint-config-airbnb-base`

 - add temp disabled `jsx-a11y/anchor-has-content` rule

ljharb committed:
[eslint config] v10.0.1

sstern6 committed:
[eslint config] [base] [breaking] Add no-plusplus in style.js and added explanation in README.

ljharb committed:
[eslint config] [deps] update `eslint-plugin-react`

ljharb committed:
[eslint config] [base] [fix] correct `import/extensions` list.

From b45fdcd60f6e08f13718bc2348e8d6ea1314e91c / fixes #1013.

Commit:
Add section on variable assignment chaining

Commit:
Update README.md

Add M2GEN to the list

ljharb committed:
[eslint config] [*] [deps] update `eslint`, `eslint-plugin-react`

lespacedunmatin committed:
Typo : text link to 13.5 was not working (#1020)

lencioni committed:
Remove misleading reference to one-var eslint rule

This eslint rule does not enforce this style guide rule. I believe that
referencing it here will cause people to believe that this is enforced
via linting when it is not. As far as I know, there is no rule that
currently enforces this, but it would be nice to add one.

soul-wish committed:
fix issue with eslint-config-airbnb README.md file formatting

ljharb committed:
ES2015

ttmarek committed:
Explain why not to prefix methods with underscores

ljharb committed:
[eslint config] [base] v5.0.3

ljharb committed:
[guide] fix “long lines” example to not include strings.

Fixes #1027.

ljharb committed:
[eslint config] [base] [patch] remove redundant `DebuggerStatement` from `no-restricted-syntax`

Fixes #1031.

hshoff committed:
[commas] fix traling comma link

hshoff committed:
[7.1] add link to fn names discussion. fixes #794

heejongahn committed:
Fix a typo in commas section

Example code block following blockquote(line 2223) was broken as a line break was missing. So I just added single blank line.

ljharb committed:
[eslint config] [patch] set `ignoreCase` to `true` in disabled rules.

When/if these rules are enabled, we will want to ignore case.

This also helps https://github.com/pinterest-web/eslint-config-pinterest/pull/46/files#r76165293

lencioni committed:
Add CSS-in-JavaScript section

We have been writing CSS in JavaScript and have developed a set of best
practices around this type of thing. I thought it would be nice to
publish this document alongside our other JavaScript style guide
documents.

lencioni committed:
Re-order breakpoints in CSS-in-JS document

This is more "mobile first".

ljharb committed:
[guide] remove subtle endorsement of JSDoc-style commenting.

lencioni committed:
Add note about where where mystery objects come from

As @ljharb pointed out, we are using some functions like `css()` and
`withStyles()`, but make no mention of what they are or where they come
from. Since they come from react-with-styles, let's make this clear.

ljharb committed:
Merge pull request #1047 from airbnb/css-in-js

[guide] Add CSS-in-JavaScript section

captbaritone committed:
Document react/require-render-return as a fully fledged rule

Now that the pull request has been merged, we can reference the actual rule.

ljharb committed:
[eslint config] [base] [breaking] enable `no-template-curly-in-string`

 - Also, clean up rule documentation.

ljharb committed:
[eslint config] [base] [breaking] enable `func-call-spacing`

ljharb committed:
[eslint config] [base] [breaking] enable `no-tabs`

ljharb committed:
[eslint config] [base] [breaking] enable `no-bitwise`

cshg committed:
Fix header link to Classes & Constructors from ECMAScript 6+ Styles

ljharb committed:
[eslint config] [base] [deps] update `eslint`, `eslint-find-rules`, `eslint-plugin-import`

ljharb committed:
[eslint config] [base] [breaking] enable `symbol-description`

ljharb committed:
[eslint config] [base] [breaking] enable `class-methods-use-this`

ljharb committed:
[eslint config] [base] [breaking] `valid-typeof`: enable `requireStringLiterals` option

ljharb committed:
[eslint config] [base] Update `ecmaVersion` to `2016`.

ljharb committed:
[eslint config] [base] v6.0.0

ljharb committed:
[eslint config] [*] [docs] use “#” in example command rather than version numbers.

Closes #984.

ljharb committed:
[eslint config] [base] v7.0.0

ljharb committed:
[eslint config] [deps] update `eslint-config-airbnb-base`, `eslint`, `eslint-find-rules`, `eslint-plugin-import`, `eslint-plugin-jsx-a11y`, `eslint-plugin-react`

ljharb committed:
[eslint config] [breaking] enable `jsx-a11y` rules:

 - `jsx-a11y/anchor-has-content`
 - `jsx-a11y/tabindex-no-positive`
 - `jsx-a11y/no-static-element-interactions`

ljharb committed:
[eslint config] [breaking] enable `react` rules:

 - `react/no-danger-with-children`
 - `react/no-unused-prop-types`
 - `react/style-prop-object`
 - `react/forbid-prop-types`
 - `react/jsx-no-duplicate-props`

Also sets `react/no-danger` to “warn”.

ljharb committed:
[guide] Update declarations section to prefer expressions.

Fixes #1008.

ljharb committed:
[eslint config] v11.0.0

lencioni committed:
Update ECMAScript 6 heading to match TOC

I plan to add some information to this section that applies to ES6 and
everything following ES6. Since we already list this as ES6+ in the
table of contents, I figured it would be a good idea to make this match.

I preserved the anchor to avoid breaking links for folks.

lencioni committed:
Add rule about TC39 proposals that haven't reached stage 3

There has been some confusion around which language features should be
used and which should not be used yet. We have a pretty clear-cut rule
at Airbnb about this, which is do not use any TC39 proposals that have
not reached stage 3.

I considered putting this at the very top of the document, but I spotted
this section that seemed to fit.

Addresses #1057

lencioni committed:
Add JavaScript Air to list of Podcasts

It probably wouldn't be a bad idea for us to make a full pass on
updating the resources section, but I noticed the podcasts portion was
missing JS Air.

lencioni committed:
Prefer https over http

I checked all of the links in the JS style guide document and converted
as many as I could to use https.

ljharb committed:
[eslint config] [base] [deps] update to `eslint` `v3.5.0`.

ljharb committed:
[eslint config] [base] [patch] loosen `max-len` by ignoring strings

Per https://github.com/eslint/eslint/pull/7049 / https://github.com/eslint/eslint/issues/5805

ljharb committed:
[eslint config] [base] v7.0.1

ljharb committed:
[eslint config] [base] [minor] enable `arrow-parens` rule

This will cause new failures for some, but only for those not following our style guide already - hence, it’s not semver-major.

ljharb committed:
[eslint config] [base] v7.1.0

ljharb committed:
[eslint config] [deps] update `eslint-config-airbnb-base`, `eslint`

ljharb committed:
[eslint config] v11.1.0

Doweig committed:
[Docs] update 8.3 to fix conflict with 6.2 (fixes #1058)

alexeyraspopov committed:
move #variables--unary-increment-decrement to correct section

Doweig committed:
[guide] Fix typo from #1058

ljharb committed:
[eslint config] [base] [patch] fix category of `no-restricted-properties`

Per https://github.com/eslint/eslint/issues/7112 / https://github.com/eslint/eslint/issues/7118

kesne committed:
[guide] Add notes on acronyms and initialisms to naming

ljharb committed:
[eslint config] [*] [deps] update `eslint-plugin-import`

ljharb committed:
[eslint config] [deps] update `eslint-plugin-jsx-a11y`, `eslint-plugin-react`

yamadashy committed:
Fix japanese translation link

ljharb committed:
[Docs] remove yet another wrapped-string "good" example

lencioni committed:
Add rule to avoid misusing DOM component prop names

People expect props like `style` and `className` to mean one specific
thing. Varying this API for a subset of your app makes the code less
readable and less maintainable, and may cause bugs.

lencioni committed:
Merge pull request #1083 from airbnb/react-props

Add rule to avoid misusing DOM component prop names

ljharb committed:
[eslint config] [*] [deps] update `eslint-find-rules`, `safe-publish-latest`

ljharb committed:
[eslint config] [deps] update `eslint-plugin-react`

ljharb committed:
[eslint config] [*] [new] set `ecmaVersion` to 2017; enable object rest/spread; update `babel-preset-airbnb`

ljharb committed:
[eslint config] [*] [deps] update `eslint-plugin-import`, `eslint` to 3.6

ljharb committed:
[eslint config] [base] v7.2.0

ljharb committed:
[eslint config] [deps] update `eslint-config-airbnb-base`

ljharb committed:
[eslint config] v11.2.0

ljharb committed:
[eslint config] [base] [breaking] enable rules:

 - `no-restricted-properties`
 - `prefer-numeric-literals`
 - `lines-around-directive`
 - `import/extensions`
 - `import/no-absolute-path`
 - `import/no-dynamic-require`

ljharb committed:
[eslint config] [base] v8.0.0

ljharb committed:
[eslint config] [breaking] [deps] update `eslint-config-airbnb-base`

ljharb committed:
[eslint config] [patch] disable deprecated and redundant `react/require-extension` rule

Fixes #978.

ljharb committed:
[eslint config] [breaking] Enable react rules:

 - `react/no-unescaped-entities`
 - `react/no-children-prop`

ljharb committed:
[eslint config] v12.0.0

ljharb committed:
[eslint config] [patch] `class-methods-use-this`: exempt React lifecycle methods

jonase committed:
[eslint config] [base] [patch] `new-cap`: add immutable.js exceptions

vdh committed:
[eslint config] [patch] `class-methods-use-this`: exempt React `getChildContext`

Added `getChildContext` to the exceptions list for `class-methods-use-this`

jcreamer898 committed:
Lonely Planet

Add Lonely Planet to the "In the Wild" list.

SimenB committed:
Upgrade to import plugin v2

ljharb committed:
[eslint config] [*] update `babel-preset-airbnb`, `eslint`, `eslint-find-rules`, `tape`

Commit:
Make section numbering consistent

* Renumber items where numbers were skipped

karanjthakkar committed:
Add section on multi-line imports. Fixes #1051

karanjthakkar committed:
Remove legacy anchor link. Add dangling comma

karanjthakkar committed:
Fix rule description

ljharb committed:
Merge pull request #1110 from karanjthakkar/multiline-imports

[guide] Add section on multiline imports.

Fixes #1051

randallreedjr committed:
Update comparison recommendation

* Use shortcuts for boolean comparisons
* Use explicit comparisons for strings and numbers

chesterhow committed:
Fix naming typo

ljharb committed:
[eslint config] [*] [deps] update `safe-publish-latest`, `eslint-plugin-jsx-a11y`, `eslint-plugin-react`

milesburton committed:
Update README.md

Minor oversight generating map object. Need square brackets to generate dynamic key.

ljharb committed:
[eslint config] [deps] update `eslint-plugin-react`

SimenB committed:
Enable `import/no-webpack-loader-syntax` rule

ljharb committed:
[eslint config] [base] Add comments on new import rules.

ljharb committed:
[eslint config] [*] [docs] ensure latest version of config is installed

Fixes #1121.

ljharb committed:
[guide] improve ++/-- example.

Fixes #1130.

SimenB committed:
Update eslint to 3.8.0

New rule called `func-name-matching` added

mifi committed:
Use import instead of require

Don't recommend the use of require
(See 10.1)

SimenB committed:
Add `ForOfStatement` to `no-restricted-syntax`

Fixes #1122

ljharb committed:
[eslint config] [base] v9.0.0

ljharb committed:
[eslint config] [deps] [breaking] update `eslint-config-airbnb-base`, `eslint-plugin-import`

ljharb committed:
[eslint config] [patch] set `react/no-unused-prop-types` to “warn”, pending bugfixes.

Closes #1099.

ljharb committed:
[eslint config] [*] [deps] update `eslint`

AndrewSouthpaw committed:
[guide] Fix inconsistency in function expression rules.

spen committed:
Add links to mentioned github issues

1pete committed:
[react] add rule references in section *Spacing*

1pete committed:
Fix broken rule reference link `import/first`

1pete committed:
Add reference for `Object.assign`

Commit:
Merge branch 'master' of https://github.com/airbnb/javascript into feature/update

ljharb committed:
[eslint config] [base] [deps] [breaking] update `eslint`

 - enable `no-useless-return`
 - tighten up `indent`
 - loosen `max-len` with `ignoreRegExpLiterals` option

ljharb committed:
[eslint config] [*] [dev deps] update `eslint-find-rules`

ljharb committed:
[eslint config] [base] correct typo from a23a3efba9d1661e52cfa6894d620cee965ac090

ljharb committed:
[eslint config] [deps] update `eslint`

ljharb committed:
[eslint config] [base] correct another typo from a23a3efba9d1661e52cfa6894d620cee965ac090

dan-h-ch committed:
Updated README.md - Corrected capitalize on GitHub

nischayv committed:
Updated README - Added sub-section for spaces in comments

gianpaj committed:
DailyJS is not longer being published

ljharb committed:
[Docs] add reference to spaced-comment rule; flesh out options

dijonkitchen committed:
Create LICENSE.md

graingert committed:
[eslint config] [base] [breaking] `comma-dangle`: require trailing commas for functions

KevinGrandon committed:
Update JSX quote style rationale.

Based on https://github.com/airbnb/javascript/issues/629#issuecomment-166384924.

Commit:
add quri files

Commit:
delete files

Commit:
bump version

BusbyActual committed:
[guide] quotes: corrected casing

Updated case to read the same in multiple spots. Added single quote example

ljharb committed:
[guide] add section on multiline function signatures/invocations

vanillajonathan committed:
[guide] Mention both num++ and num ++

chollier commented on packages/eslint-config-quri-base/package.json:
```diff
@@ -44,17 +44,21 @@
   },
   "peerDependencies": {
-    "eslint": "^3.0.0",
-    "eslint-plugin-babel": "^3.3.0",
```
no more eslint-plugin-babel?

chollier commented on packages/eslint-config-quri-base/rules/best-practices.js:
```diff
@@ -63,161 +71,188 @@ module.exports = {
+
+    // disallow implicit type conversions
+    // http://eslint.org/docs/rules/no-implicit-coercion
+    'no-implicit-coercion': ['off', {
```
is this intended?

chollier commented on packages/eslint-config-quri-base/rules/imports.js:
```diff
@@ -15,44 +15,54 @@ module.exports = {
+    },
+    'import/extensions': [
+      '.js',
+      '.jsx',
```
not sure we need jsx

chollier commented on packages/eslint-config-quri-base/rules/style.js:
```diff
@@ -67,78 +79,111 @@ module.exports = {
-      ignoreComments: false
+      ignoreComments: false,
+      ignoreStrings: true,
+      ignoreTemplateLiterals: true,
```
this is great

chollier commented on packages/eslint-config-quri-base/test/README.md:
```diff
@@ -0,0 +1,59 @@
+# eslint-config-airbnb-base
```
need to rename here

chollier commented on packages/eslint-config-quri/package.json:
```diff
@@ -38,19 +38,24 @@
+    },
+    "peerDependencies": {
+    "eslint": "^3.8.1",
+    "eslint-plugin-jsx-a11y": "^2.2.3",
```
we don't need this one

alfondotnet replied:
👍

chollier commented on packages/eslint-config-quri/rules/react.js:
```diff
@@ -218,29 +235,78 @@ module.exports = {
+
+    // only .jsx files may have JSX
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-filename-extension.md
+    'react/jsx-filename-extension': ['error', { extensions: ['.jsx'] }],
```
hmm?

alfondotnet replied:
:+1:

chollier reviewed (commented):
I like that they change 0 / 1 / 2 to 'off' / 'error' / 'never' 

left a few comments, thanks for doing this 🙇

alfondotnet commented:
Thanks a lot for the comments, after a while I got a little crazy but later in the day I will fix these

negativetwelve committed:
[guide] Separate 19.2 code blocks into diff + javascript sections

ljharb committed:
[eslint config] [base] [breaking] update `eslint`, `eslint-plugin-import`; enable `import/no-named-default`

ljharb committed:
[eslint config] [base] [breaking] prefer `**` over `Math.pow`

ljharb committed:
[eslint config] [base] [patch] loosen `no-extraneous-dependencies` for test files.

Relates to #959, #1089.

ljharb committed:
[Tests] on `node` `v7`

ljharb committed:
[eslint config] [base] v10.0.0

spen committed:
[guide] Update syntax in `eol-last` examples

ljharb committed:
[eslint config] [breaking] [deps] update `eslint-config-airbnb-base`, `eslint`, `eslint-plugin-import`, `eslint-plugin-react`

ljharb committed:
[eslint config] [breaking] re-enable `react/no-unused-prop-types`

ljharb committed:
[eslint config] v13.0.0

ljharb committed:
[eslint config] [base] [fix] legacy config should not require `**`

Per https://github.com/airbnb/javascript/commit/1399b3cbfa785f760c15ec63112f711b2c68fd28#commitcomment-19718532

ljharb committed:
[eslint config] [base] v10.0.1

ljharb committed:
[eslint config] [deps] update `eslint-config-airbnb-base`

Commit:
bring back eslint-plugin-babel

Commit:
remove jsx from import extension rules

Commit:
rename airbnb -> quri

Commit:
remove unnecessary peer dependency

Commit:
airbnb -> quri

Commit:
remove wrong jsx rule

chentsulin committed:
[eslint config] [base] `import/no-extraneous-dependencies`: added ignore patterns for config files

Fixes #1168

ljharb committed:
[guide] [react] add note that defaultProps must always be provided for non-required props

chollier commented on packages/eslint-config-quri/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-quri",
-  "version": "3.0.1",
+  "version": "3.0.2",
```
I would do a major bump here

chollier commented on packages/eslint-config-quri/test/package.json:
```diff
@@ -41,14 +41,14 @@
   },
   "devDependencies": {
-    "babel-preset-airbnb": "^2.1.1",
+    "babel-preset-quri": "^2.1.1",
```
we don't have that package

chollier committed:
Merge remote-tracking branch 'upstream/master' into feature/update

# Conflicts:
#	README.md
#	packages/eslint-config-quri-base/package.json
#	packages/eslint-config-quri-base/rules/style.js
#	packages/eslint-config-quri-base/test/rules/imports.js
#	packages/eslint-config-quri/test/package.json
#	react/README.md

chollier committed:
more updates
